Lib/zipimport.py

```diff
@@ -353,17 +353,43 @@ def _read_directory(archive):
         raise ZipImportError(f"can't open Zip file: {archive!r}", path=archive)
 
     with fp:
+        end_cent_dir_size = 22
         try:
-            fp.seek(-22, 2)
+            fp.seek(-end_cent_dir_size, 2)
             header_position = fp.tell()
-            buffer = fp.read(22)
+            buffer = fp.read(end_cent_dir_size)
         except OSError:
             raise ZipImportError(f"can't read Zip file: {archive!r}", path=archive)
-        if len(buffer) != 22:
+        if len(buffer) != end_cent_dir_size:
             raise ZipImportError(f"can't read Zip file: {archive!r}", path=archive)
-        if buffer[:4] != b'PK\x05\x06':
+        string_end_archive = b'PK\x05\x06'
```

Comment:
This is another candidate for a module constant.

Reply:
Done.